### PR TITLE
nixos/make-options-doc: skip re-escaping literal expressions

### PR DESCRIPTION
**Pull request by sandydoo**
###### Description of changes

Modifies the CommonMark and AsciiDoc generators to not re-escape literal expressions and other literal types. These strings are escaped when serialized to JSON and should just be unpacked.

@domenkozar 

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by domenkozar** [comment]
cc @roberth

**Comment by sandydoo** [comment]
I should probably move the object hook logic into the encoder.

File: nixos/lib/make-options-doc/generateDoc.py [hunk 1]
```diff
@@ -49,14 +53,14 @@ def generate_commonmark(options):
         if 'default' in value:
             print('*_Default_*')
             print('```')
-            print(json.dumps(value['default'], ensure_ascii=False, separators=(',', ':')))
+            print(json.dumps(value['default'], cls=OptionsEncoder, ensure_ascii=False, separators=(',', ':')))
```
Comment (roberth) [inline nixos/lib/make-options-doc/generateDoc.py:56]: This is still wrong. It shouldn't use a json encoder at all, but rather pretty print a Nix value.

**Comment by roberth** [comment]
cc @ncfavier because I have no idea whether this impacts search.nixos.org.

**Comment by roberth** [comment]
We're going to need tests for this code. It has too many hacks to just change things without any checks.

**Comment by ncfavier** [comment]
> cc @ncfavier because I have no idea whether this impacts search.nixos.org.

Nope, we don't use the MD/AD generators, only optionsJSON.

File: nixos/lib/make-options-doc/generateDoc.py (hunk 1 above)
Comment (sandydoo) [reply nixos/lib/make-options-doc/generateDoc.py:56]: Ah, cool! I was worried there could be arbitrary JSON in there, given all these json dumps, but looks like we don't need any of that.

**Comment by sandydoo** [comment]
> We're going to need tests for this code. It has too many hacks to just change things without any checks.

@roberth, agreed! This would be a good spot to check the output of this, right?

https://github.com/NixOS/nixpkgs/blob/0f538a65499f3668d04e4723040b716319ceb0db/nixos/modules/misc/documentation/test.nix#L40-L49

**Comment by roberth** [comment]
Yes, that looks like the place to do it. It would be nice if we could make ofborg could pick that up...

**Pull request merged**